compiler/rustc_next_trait_solver/src/solve/eval_ctxt/canonical.rs

```diff
@@ -102,8 +97,8 @@ impl<'tcx> EvalCtxt<'_, InferCtxt<'tcx>> {
 
         // We only check for leaks from universes which were entered inside
         // of the query.
-        self.infcx.leak_check(self.max_input_universe, None).map_err(|e| {
-            trace!(?e, "failed the leak check");
+        self.infcx.leak_check(self.max_input_universe).map_err(|NoSolution| {
```

Comment:
Actually, `leak_check` returns `TypeError` which I *did* previously uplift. I changed this without thinking -- could leave a FIXME to restore it.

Reply:
please add a FIXME to uplift the leak check instead.